src/main/resources/avro/genotypephenotypemethods.avdl

```diff
@@ -0,0 +1,319 @@
+  /**
+  The `Dataset` to search.
+  */
+  string datasetId;
+
+  /**
+  Specifies the maximum number of results to return in a single page.
+  If unspecified, a system default will be used.
+  */
+  union { null, int } pageSize = null;
+
+  /**
+  The continuation token, which is used to page through large result sets.
+  To get the next page of results, set this parameter to the value of
+  `nextPageToken` from the previous response.
+  */
+  union { null, string } pageToken = null;
+}
+
+/** This is the response from `POST /phenotypeassociationsets/search` expressed as JSON. */
+record SearchPhenotypeAssociationSetsResponse {
+
+  /** The list of matching phenotype association sets. */
+  array<org.ga4gh.models.PhenotypeAssociationSet> phenotypeAssociationSets = [];
+
+  /**
+  The continuation token, which is used to page through large result sets.
+  Provide this value in a subsequent request to return the next page of
+  results. This field will be empty if there aren't any additional results.
+  */
+  union { null, string } nextPageToken = null;
+}
+
+/**
+Gets a list of `PhenotypeAssociationSet`s matching the search criteria.
+
+`POST /phenotypeassociationsets/search` must accept a JSON version of
+`SearchPhenotypeAssociationSetsRequest` as the post body and will return a JSON version
+of `SearchPhenotypeAssociationSetsResponse`.
+*/
+SearchPhenotypeAssociationSetsResponse searchPhenotypeAssociationSets(
+  /** This request maps to the body of `POST /phenotypeassociationsets/search` as JSON. */
+  SearchPhenotypeAssociationSetsRequest request) throws GAException;
+
+
+/******************  /phenotypes/search  *********************/
+
+/**
+One or more phenotypes can be queried together.
+*/
+record PhenotypeQuery {
+  union { null, string } name; /* new field, regex */
+  union { null, org.ga4gh.models.OntologyTerm } type = null;
+  union { null, array<org.ga4gh.models.OntologyTerm> } qualifier = null;
+  union { null, org.ga4gh.models.OntologyTerm } ageOfOnset = null;
+  union { null, string } description = null;  /*regex*/
+  union { null, array<org.ga4gh.models.ExternalIdentifier> }  externalIdentifiers = null;
+}
+
+/**
+One or more ontology terms can be queried together.
+*/
+record OntologyTermQuery {
+  array<org.ga4gh.models.OntologyTerm> terms;
+}
+
+/**
+One or more ids can be queried together.  Generally used for instances
+of a particular class of object (e.g. a specific gene or SNP).
+*/
+record ExternalIdentifierQuery {
+  array<org.ga4gh.models.ExternalIdentifier> ids;
+}
+
+/**
+Evidence for the phenotype association.
+*/
+record EvidenceQuery {
+  /** ECO or OBI is recommended */
+  union { null, org.ga4gh.models.OntologyTerm }  evidenceType = null;
+  union { null, string } description = null;  /*regex*/
+  union { null, array<org.ga4gh.models.ExternalIdentifier> }  externalIdentifiers = null; /* new field */
+}
+
+
+record GenotypeQuery {
+  /**
+  only those fields from feature that are `queryable`
```

Comment:
Yes, the semantics of the original query 
```
The goal here is to allow users to query using one or more of
Genotype, Phenotype, Environment, and Evidence.

A query using one of the above items is to mean, by default,
that the remainder of the query is as a "wildcard", such
that all matches to just that query term would come back.
Combinations of the above are to act like AND rather than OR.
```